### PR TITLE
autobuild: add working-directory input

### PR DESCRIPTION
**Pull request by cklin**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by cklin** [comment]
@igfoo Do you have any thoughts on how we should test this functionality?

File: src/autobuild-action.ts [hunk 1]
```diff
@@ -71,6 +72,13 @@ async function run() {
     }
     language = determineAutobuildLanguage(config, logger);
     if (language !== undefined) {
+      const workingDirectory = getOptionalInput("working-directory");
+      if (workingDirectory) {
+        logger.info(
+          "Changing autobuilder working directory to ${workingDirectory}"
```
Comment (aeisenberg) [inline src/autobuild-action.ts:78]: You need backticks here:

```suggestion
          `Changing autobuilder working directory to ${workingDirectory}.`
```

Comment (cklin) [reply src/autobuild-action.ts]: Fixed.  Thank you for catching this!

**Comment by aeisenberg** [comment]
Looks like a spurious failure.

**Comment by aeisenberg** [comment]
Best way to test this would be to generate a small test project that uses the autobuild action. Create a workflow that runs codeql on that project and make sure to specify a working directory input. Perhaps something similar to the `checkout_path` tests.

**Comment by igfoo** [comment]
I've given it a try, and it looks like it works for me; thanks!

**Review by aeisenberg — APPROVED** [review]
nice

**Comment by cklin** [comment]
Finally got `PR Check - Autobuild working directory` to work:
- Check passing with `working-directory` input: https://github.com/github/codeql-action/actions/runs/2117821927
- Check failing without `working-directory` input: https://github.com/github/codeql-action/actions/runs/2117807488

**Pull request merged**